akka-http/src/main/scala/akka/http/javadsl/server/directives/BasicDirectives.scala

```diff
@@ -327,4 +327,16 @@ abstract class BasicDirectives {
     D.toStrictEntity(timeout) { inner.get.delegate }
   }
 
+  /**
+   * WARNING: This will read the entire request entity into memory and effectively disable streaming.
+   *
+   * Extracts the [[akka.http.javadsl.server.RequestContext]] itself with the strict HTTP entity,
+   * or fails the route if unable to drain the entire request body within the timeout.
```

Comment:
I think this one is tricky, as it fails the request the same way as timeout, there was some recent discussions around this described in #2184, so it does fail the request but not quite the `route`

Reply:
> as it fails the request the same way as timeout,

Why do you think so? Wouldn't it just fail in the route right here?

The low-level request timeouts instead are handled on the core layer and never reach the higher layer.

Reply:
Ah this might be fine indeed: whoever consumes the stream (typically some unmarshaller) will fail, which will then e.g. lead to an `MalformedRequestContentRejection` and be processed like usual.

Reply:
Ok, thanks for clarifying!